File: app/src/main/java/co/edu/uniautonoma/posgradosapp/Actividades/PosgradosActivity.java
package co.edu.uniautonoma.posgradosapp.Actividades;

import android.content.Intent;
import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;

import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import java.util.ArrayList;

import co.edu.uniautonoma.posgradosapp.Modelos.Posgrados;
import co.edu.uniautonoma.posgradosapp.R;
import co.edu.uniautonoma.posgradosapp.ViewModels.PosgradosViewModel;
import xdroid.toaster.Toaster;

public class PosgradosActivity extends AppCompatActivity {

    private ArrayList<Posgrados> posgrados = new ArrayList<>();
    private ArrayList<String> listaposgrados = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_seleccionar_posgrado);
        PosgradosViewModel viewModel = ViewModelProviders.of(this).get(PosgradosViewModel.class);
        if(viewModel.getAllPosgrados()!=null){
            viewModel.getAllPosgrados().observe(this, posgrado ->{
                posgrados = posgrado;
                llenarPosgrados();
            });
        }else {
            Toaster.toast(R.string.EstadoServidor);
        }
    }

    private void llenarPosgrados() {

        ListView lvPosgrados = (ListView) findViewById(R.id.lvPosgrados);

        for(int i = 0; i< posgrados.size(); i++) {
            listaposgrados.add(posgrados.get(i).getNombre());
        }
        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(this, R.layout.lista_posgrado, R.id.tvlistaposgrado, listaposgrados);
        lvPosgrados.setAdapter(arrayAdapter);

        lvPosgrados.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Intent i = new Intent(PosgradosActivity.this, DetallePosgradoActivity.class);
                i.putExtra("id_posgrado", posgrados.get(position).getId());
                i.putExtra("cod_snies", posgrados.get(position).getCod_snies());
                i.putExtra("nombre", posgrados.get(position).getNombre());
                i.putExtra("duracion", posgrados.get(position).getDuracion());
                i.putExtra("totalcreditos", posgrados.get(position).getTotalcreditos());
                i.putExtra("descripcion", posgrados.get(position).getDescripcion());
                i.putExtra("valorsemestre", posgrados.get(position).getValorsemestre());
                startActivity(i);
            }
        });
    }
}

File: presentation/src/main/java/co/edu/uniautonoma/posgradosapp/presentation/ui/posgrados/PosgradosActivity.kt
package co.edu.uniautonoma.posgradosapp.presentation.ui.posgrados

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.lifecycle.Observer
import co.edu.uniautonoma.posgradosapp.presentation.R
import co.edu.uniautonoma.posgradosapp.domain.entity.Posgrados
import co.edu.uniautonoma.posgradosapp.presentation.ui.base.BaseActivity
import kotlinx.android.synthetic.main.activity_seleccionar_posgrado.*
import org.koin.androidx.viewmodel.ext.android.viewModel
import xdroid.toaster.Toaster
import java.util.*

class PosgradosActivity : BaseActivity() {

    private val posgradosViewModel: PosgradosViewModel by viewModel()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_seleccionar_posgrado)

        mostrarDialog()
        observarResultado()
    }

    private fun observarResultado() {

        posgradosViewModel.posgradosLiveData.observe(this, Observer {
            llenarPosgrados(it)
            mostrarPosgrado(it)
            ocultarDialog()
        })

        posgradosViewModel.errorLiveData.observe(this, Observer {
            ocultarDialog()
            Toaster.toast(R.string.EstadoServidor)
            Log.d("PosgradosError:", it)
        })
    }

    private fun llenarPosgrados(posgrados: List<Posgrados>) {
        val listaposgrados = ArrayList<String>()
        posgrados.map {
            it.nombre?.let {
                listaposgrados.add(it)
            }
        }
        lvPosgrados.adapter = ArrayAdapter(this, R.layout.lista_posgrado, R.id.tvlistaposgrado, listaposgrados)
    }

    private fun mostrarPosgrado(posgrados: List<Posgrados>) {
        lvPosgrados.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            val i = Intent(this@PosgradosActivity, DetallePosgradoActivity::class.java)
            i.putExtra("id_posgrado", posgrados[position].id)
            i.putExtra("cod_snies", posgrados[position].cod_snies)
            i.putExtra("nombre", posgrados[position].nombre)
            i.putExtra("duracion", posgrados[position].duracion)
            i.putExtra("totalcreditos", posgrados[position].totalcreditos)
            i.putExtra("descripcion", posgrados[position].descripcion)
            i.putExtra("valorsemestre", posgrados[position].valorsemestre)
            startActivity(i)
        }
    }
}